src/resource_clients/request_queue.ts

```diff
@@ -96,6 +98,111 @@ export class RequestQueueClient extends ResourceClient {
         return cast(parseDateFields(pluckData(response.data)));
     }
 
+    protected async _batchAddRequestsWithRetries(
+        requests: Omit<RequestQueueClientRequestSchema, 'id'>[],
+        options: RequestQueueClientBatchAddRequestOptions = {},
+    ): Promise<RequestQueueClientBatchAddRequestsResult> {
+        // Keep track of the requests that remain to be processed (in parameter format)
+        let remainingRequests = requests;
+        // Keep track of the requests that have been processed (in api format)
+        const processedRequests: ProcessedRequest[] = [];
+        // The requests we have not been able to process in the last call
+        // ie. those we have not been able to process at all
+        let unprocessedRequests: UnprocessedRequest[] = [];
+        for (let i = 0; i < 1 + this.httpClient.maxRetries; i++) {
```

Comment:
I would do not mix maxRetries with retries for request queue unprocessed items.
Users can set this opt on the client level and probably do not expect this side-effect.